### PR TITLE
avoid calling `get` on empty cache

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/resolver.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export default class RuntimeResolver implements IRuntimeResolver<OwnedTemplateMe
   } = BUILTIN_MODIFIERS;
 
   // supports directly imported late bound layouts on component.prototype.layout
-  private templateCache: Map<Owner, Map<TemplateFactory, OwnedTemplate>> = new Map();
```
Comment (bekzod) [inline packages/@ember/-internals/glimmer/lib/resolver.ts]: `WeakMap` is more appropriate for `templateCache` I think, as owner destroyed all cache will be destroyed along with it

File: packages/@ember/-internals/glimmer/lib/resolver.ts [hunk 2]
```diff
@@ -188,11 +188,14 @@ export default class RuntimeResolver implements IRuntimeResolver<OwnedTemplateMe
...
   createTemplate(factory: TemplateFactory, owner: Owner): OwnedTemplate {
     let cache = this.templateCache.get(owner);
+    let template;
     if (cache === undefined) {
       cache = new Map();
       this.templateCache.set(owner, cache);
+    } else {
+      template = cache.get(factory);
     }
-    let template = cache.get(factory);
```
Comment (bekzod) [inline packages/@ember/-internals/glimmer/lib/resolver.ts]: this is called even if `cache` just created

File: packages/@ember/-internals/glimmer/lib/resolver.ts (hunk 1 above)
Comment (rwjblue) [reply packages/@ember/-internals/glimmer/lib/resolver.ts]: Actually no, we absolutely want a `Map` here. This actually _resolved_ a memory leak in the 3.1 series.

We _never_ want to release templates during the lifetime of the application (which is what this resolver is scoped to).

**Review by rwjblue — CHANGES_REQUESTED** [review]
Need to revert one part, but the changes in `createTemplate` seem good

**Review by locks — APPROVED** [review]

**Comment by bekzod** [comment]
@locks can we restart this ?

**Comment by bekzod** [comment]
good to merge :)

**Review by Turbo87 — APPROVED** [review]

**Pull request merged**